pyvista/core/grid.py

```diff
@@ -852,3 +852,37 @@ def extent(self, new_extent: Sequence[int]):
     def to_tetrahedra(self, *args, **kwargs):
         """Cast to a rectangular grid and then convert to tetrahedra."""
         return self.cast_to_rectilinear_grid().to_tetrahedra(*args, **kwargs)
+
+
+class UniformGrid(ImageData):
+    """Deprecated ``UniformGrid`` and renamed to ``ImageData``.
+
+    .. deprecated:: 0.40.0
+        ``pyvista.UniformGrid`` has been deprecated and renamed to ``pyvista.ImageData``.
```

Comment:
Not sure this will get rendered anywhere, but
```suggestion
        ``pyvista.UniformGrid`` has been deprecated and renamed to :class:`pyvista.ImageData`.
```